src/boot/efi/boot.c

```diff
@@ -1162,6 +1271,35 @@ static void config_add_entry(Config *config, BootEntry *entry) {
         config->entries[config->n_entries++] = entry;
 }
 
+static void config_add_group(Config *config, BootGroup *group) {
+        assert(config);
+        assert(group);
+
+        assert(config->n_groups < IDX_MAX);
+
+        config->groups = xrealloc(
+                        config->groups,
+                        sizeof(void *) * config->n_groups,
+                        sizeof(void *) * (config->n_groups + 1));
```

Comment:
we should really add xreallocarray() that works like xrealloc() but is modelled after reallocarray() in userpace